core/src/main/java/org/springframework/security/oauth2/server/authorization/client/RegisteredClient.java

```diff
@@ -18,15 +18,29 @@
 import org.springframework.security.core.SpringSecurityCoreVersion;
 import org.springframework.security.oauth2.core.AuthorizationGrantType;
 import org.springframework.security.oauth2.core.ClientAuthenticationMethod;
+import org.springframework.util.Assert;
+import org.springframework.util.CollectionUtils;
+import org.springframework.util.StringUtils;
 
 import java.io.Serializable;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.Collection;
 import java.util.Collections;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
 import java.util.Set;
 
 /**
+ * A representation of a client registration with an OAuth 2.0 Authorization Server.
+ *
  * @author Joe Grandja
+ * @author Anoop Garlapati
+ * @since 5.3
+ * @see <a target="_blank" href="https://tools.ietf.org/html/rfc6749#section-2">Section 2 Client Registration</a>
  */
-public class RegisteredClient implements Serializable {
+public final class RegisteredClient implements Serializable {
```

Comment:
final might be too harsh.
In API's like this the developer should have some freedom that enables to change some parts of the implementation

Reply:
Fixed. I started out with final class thinking because it should be completely immutable but I agree final is harsh on top of omitting the setters.